django-testbed/frontend/src/models/User.js

```diff
@@ -13,21 +13,30 @@ import namedUrls from '../namedUrls';
 
 import BaseUser from './generated/BaseUser';
 
+const middleware = [viewModelCachingMiddleware(() => User)];
+
 const endpoints = {
-    retrieve: new ViewModelEndpoint(namedUrls.get('users-detail'), () => User),
-    update: new ViewModelEndpoint(namedUrls.get('users-detail'), () => User, {
+    retrieve: new Endpoint(namedUrls.get('users-detail'), { middleware }),
+    update: new Endpoint(namedUrls.get('users-detail'), {
+        middleware,
         method: 'patch',
     }),
-    list: new PaginatedViewModelEndpoint(namedUrls.get('users-list'), () => User, {
+    list: new Endpoint(namedUrls.get('users-list'), {
+        middleware: [...middleware, paginationMiddleware()],
         resolveUrl(urlPattern, urlArgs, query) {
             return urlPattern.resolve(urlArgs, {
                 // default to pageNumber pagination if not specified
                 query: { ...query, paginationType: query.paginationType || 'pageNumber' },
             });
         },
```

Comment:
it's only used here to add some defaults to `query` if not otherwise set (usually would do this at the view level anyway, don't remember why was done here)

Reply:
I don't quite understand -- this looks like it's just adding some default params for pagination purposes.

If you have to do that on the endpoint then won't you have to do that on every endpoint that uses pagination? And if that's the case then shouldn't it be pagination that takes care of it?

Reply:
In the testbed you choose pagination type for test purposes - page number, cursor, offset/limit - this is just setting which type to use by default. Not relevant to general case.

Reply:
Ah I misunderstood, I thought this would always be required